Make the front-end httpd image small and simple

bdunne opened the pull request:
- Base it on the CentOS7 httpd image
- Remove currently unused packages (external auth will add what it needs)
- Only launch httpd
- Remove all unnecessary compiling of UI assets
- Consolidate all httpd configuration into one file

@abellotti Please also review

carbonin commented on Dockerfile:
```diff
@@ -1,119 +1,37 @@
-                   git                     \
-                   net-tools               \
-                   nodejs                  \
                    httpd                   \
```
If we're basing on `centos/httpd` rather than a base centos image we don't need to install the httpd rpm here.

carbonin commented on Dockerfile:
```diff
@@ -1,119 +1,37 @@
-COPY docker-assets/container.data.persist /
-COPY docker-assets/appliance-initialize.sh /bin
-ADD  docker-assets/container-scripts ${CONTAINER_SCRIPTS_ROOT}
+COPY docker-assets/generate_server_cert.sh /
```
Should we put this in a real `bin` location rather than `/`?

carbonin commented on Dockerfile:
```diff
@@ -1,119 +1,37 @@
 
 ENTRYPOINT [ "entrypoint" ]
-CMD [ "/usr/sbin/init" ]
+CMD ["/run-httpd.sh"]
```
Does this work without systemd?

I thought `apachectl` used systemd under the covers.

bdunne replied:
It appears to work Fine™

abellotti replied:
why remove systemd from here ? it is needed for external-auth

abellotti replied:
is the intent to re-add it and ext-auth in a separate PR ?

carbonin replied:
Yeah, from the description:
> Remove currently unused packages (external auth will add what it needs)

carbonin commented on docker-assets/manageiq.conf:
```diff
@@ -0,0 +1,125 @@
+  SSLProxyCheckPeerExpire off
+
+  # The following redirects files must be included to handle most specific to least specific URLs.
+#Include conf.d/manageiq-redirects-cockpit
```
Rather than bookend comments we can make these more like real section titles.

bdunne commented:
@carbonin Updated based on your review.

carbonin commented:
Should we be using dumb-init here? What is pid 1 in this container?

Also, how does it respond to scaling down? Does it exit cleanly with a SIGTERM?

abellotti commented on Dockerfile:
```diff
@@ -1,119 +1,34 @@
 ## Atomic/OpenShift Labels
 LABEL name="manageiq-apache" \
       vendor="ManageIQ" \
       version="2.4.6-45" \
```
Let's drop this version label, we'll be getting out of sync with the centos/httpd  defined one as per https://github.com/CentOS/CentOS-Dockerfiles/blob/master/httpd/centos7/Dockerfile

abellotti commented on docker-assets/manageiq.conf:
```diff
@@ -0,0 +1,114 @@
+  RequestHeader unset X-Forwarded-Proto
+  RequestHeader set X-Forwarded-Proto 'https' env=HTTPS
+  <LocationMatch "^/cws/cockpi(t[^/]+|t)?/socket$">
+    ProxyPassMatch "ws://localhost:9002/cws/cockpi$1/socket"
```
is that really served by localhost ?

bdunne replied:
No, that doesn't work right now.

abellotti commented on docker-assets/manageiq.conf:
```diff
@@ -0,0 +1,114 @@
+  RequestHeader set X_FORWARDED_PROTO 'https'
+
+  ErrorLog /var/log/httpd/ssl_error.log
+  TransferLog /var/log/httpd/ssl_access.log
```
are the logs persistent ?

bdunne replied:
not at this time

bdunne committed:
Make the front-end httpd image small and simple

- Base it on the CentOS7 httpd image
- Remove currently unused packages (external auth will add what it needs)
- Only launch httpd
- Remove all unnecessary compiling of UI assets
- Consolidate all httpd configuration into one file

bdunne commented:
@carbonin Based on my testing, it didn't have any trouble scaling down to 0

carbonin commented on docker-assets/manageiq.conf:
```diff
@@ -0,0 +1,114 @@
+<Proxy balancer://evmcluster_ui/ lbmethod=bybusyness>
+  BalancerMember https://manageiq
```
Should these bits be dynamically written from a `MANAGEIQ_SERVICE_NAME` environment variable?

I feel like that might be a bit more work but is probably the "right" way to do it.

bdunne replied:
I'd like to try that in a follow-up

carbonin commented on docker-assets/manageiq.conf:
```diff
@@ -0,0 +1,114 @@
+  RewriteCond %{REQUEST_URI} !(.*)/$
+  RewriteRule ^(.*)$ $1/ [R,NE,L]
+  ProxyPass /ansibleapi/ http://localhost:54321/api/
+  ProxyPassReverse /ansibleapi/ http://localhost:54321/api/
```
This can all be removed. Because these configs are container specific we can take advantage of the fact that ansible will be a separate service.

bdunne replied:
Can I fix it in a follow-up once your ansible PR is merged?

carbonin reviewed (approved):
Looks good.

Much easier to build this image up from essentials than have all the ManageIQ cruft here from the start.
